### PR TITLE
contig_taxonomic_assign.py updated

### PR DESCRIPTION
**Pull request by hoelzer**
@mberacochea, @guille0387 did changes to the voting system that we will now also use in the final manuscript version. 

I think it's best to merge this into `dev` and then you can merge this into `master` (or before into `cwl-docker`)? 

Some of the other branches that are currently open (`no-pprmeta`, `pprmeta-score`, `no-bitscore-filter`) where more for testing purposes and need not be merged (now). 

Would be great if we can get a new release at some point! :) 

btw: my EBI acc (email) is deactivated so I don't have any access to slack anymore. But please feel free to contact me via email or github issues ;)

### CODE REVIEW

**Comment by mberacochea** [comment]
Sure I'll review this and then merge into `dev`. I've been busy with other stuff. I'll get that you guys

File: bin/contig_taxonomic_assign.py [hunk 1]
```diff
@@ -17,105 +18,37 @@ def contig_tax(annot_df, ncbi_db, min_prot, prop_annot, tax_thres):
...
-                            if len(max_tax) > 1:
-                                assigned_taxa.append("-".join(max_tax))
-                            else:
-                                assigned_taxa.append(max_tax[0])
-        df_rows.append(assigned_taxa)
-    final_df = pd.DataFrame(
-        df_rows, columns=["contig_ID", "genus", "subfamily", "family", "order"])
-    return final_df
+                        best_lineage = ncbi.get_lineage(best_hit[0])
+                        contig_lineage.extend([ncbi.get_taxid_translator([key])[key] if pd.notnull(key) else "" for key in [{y:x for x,y in ncbi.get_rank(best_lineage).items()}.get(item) for item in tax_rank_order[tax_rank_order.index(rank):]]])
```
Comment (mberacochea) [inline bin/contig_taxonomic_assign.py:49]: Can we split this oneliner?. It's a bit difficult to read IMO.

**Review by mberacochea — APPROVED** [review]
I think we can merge this. I added just one comment for the contig_taxonomic_assign script but I can make that myself afer the merge.

**Comment by hoelzer** [comment]
> I think we can merge this. I added just one comment for the contig_taxonomic_assign script but I can make that myself afer the merge.

Yep, I think no problem to split that one-liner to make the script more readable. If you can do the merge and a new release, we can add the release version number to the current manuscript! thx!

**Pull request merged**